src/hotspot/share/code/SCCache.hpp

```diff
@@ -534,15 +544,22 @@ class SCCache : public CHeapObj<mtCode> {
   bool write_debug_info(DebugInformationRecorder* recorder);
   bool write_oop_maps(OopMapSet* oop_maps);
 
+  bool write_oop_map_set(nmethod* nm);
+  bool write_nmethod_reloc_immediates(GrowableArray<oop>& oop_list, GrowableArray<Metadata*>& metadata_list);
+  bool write_nmethod_extra_relocations(nmethod* nm, GrowableArray<oop>& oop_list, GrowableArray<Metadata*>& metadata_list);
+
```

Comment:
`write_nmethod_extra_relocations` name doesn't really say what the extra relocs are for i.e. to ensure that at load time we correctly relocate a reference into the current runtime. Could we rename this to `write_nmethod_loadtime_relocations`.